pkg/config/utils/endpoints.go

```diff
@@ -90,29 +88,98 @@ func GetMainEndpointBackwardCompatible(c pkgconfigmodel.Reader, prefix string, d
 	return prefix + pkgconfigsetup.DefaultSite
 }
 
+// MakeEndpoints takes a map of domain to apikeys and a config path root and converts this to
+// a map of domain to Endpoint structs.
+func MakeEndpoints(endpoints map[string][]string, root string) map[string][]Endpoint {
+	result := map[string][]Endpoint{}
+	for url, keys := range endpoints {
+		// Remove any empty API keys.
+		// We don't need to hold on to an endpoint with an empty API key to track if a
+		// secret has been updated since secrets can never be empty in the first place.
+		trimmed := []string{}
+		for _, key := range keys {
+			trimmedAPIKey := strings.TrimSpace(key)
+			if trimmedAPIKey != "" {
+				trimmed = append(trimmed, trimmedAPIKey)
+			}
+		}
+
+		if len(trimmed) > 0 {
+			result[url] = []Endpoint{{
+				ConfigSettingPath: root,
+				APIKeys:           trimmed,
+			}}
```

Comment:
Would it make sense to keep the info message that we skipped a url due to an empty api key (the one on old line 71).